addons/portal/static/src/js/portal_signature.js

```diff
@@ -100,30 +133,47 @@ odoo.define('portal.signature_form', function (require){
                 }
             });
         },
-
-        //--------------------------------------------------------------------------
-        // Private
-        //--------------------------------------------------------------------------
-
-        /**
-         * @private
-         * @returns {Deferred}
-         */
-        _loadTemplates: function () {
-            return ajax.loadXML('/portal/static/src/xml/portal_signature.xml', qweb);
-        },
     });
 
     base.ready().then(function () {
         $('.o_portal_signature_form').each(function () {
+            var hasBeenReset = false;
             var $elem = $(this);
-            var form = new SignatureForm(null, $elem.data());
+
+            var callUrl = $elem.data('call-url');
+            var nameAndSignatureOptions = {
+                defaultName: $elem.data('default-name'),
+                mode: $elem.data('mode'),
+                signatureRatio: $elem.data('signature-ratio'),
+                signatureType: $elem.data('signature-type'),
+            };
+            var sendLabel = $elem.data('send-label');
+
+            var form = new SignatureForm(rootWidget, {
```

Comment:
I know this is to be replaced by me afterwards but you cannot do this like that. By requiring `root.widget`, you are making the whole module (and so the widget definition) wait for the root widget to be fully initialized (dom ready, session, etc). And thus, making this code in `base.ready()` is useless.

What you have to do, temporarily, is making this `SignatureForm` initialization in its own JS module (same file) and there you can require `root.widget` and the `SignatureForm` (and no need of base then).

Reply:
Ok done, I think.